### PR TITLE
feat: setup `Prettier`

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
- related: #5015
- This is the first step of https://github.com/determined-ai/determined/pull/5015#issuecomment-1248376734

## Test Plan
- Check if Prettierrc.js file is shared
- Check if rules in eslint is correct https://github.com/determined-ai/determined/pull/5015#issuecomment-1247309010
- Check if there's nothing I missed

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/package.json
Comment (keita-determined) [inline webui/react/package.json:24]: I split up prettier command, not sure if its useful tho

File: webui/react/src/shared/configs/.prettierrc.js
Comment (keita-determined) [inline webui/react/src/shared/configs/.prettierrc.js:1]: share `prettierrc` file

File: webui/react/package.json
Comment (ClaireNeveu) [reply webui/react/package.json:24]: The one suggestion I have here is to use the generic `format` instead of "prettier"

**Review by ClaireNeveu — APPROVED** [review]

File: webui/react/package.json
Comment (ClaireNeveu) [reply webui/react/package.json:24]: Approved after this change

**Comment by hamidzr** [comment]
to make sure we're running the same thing: after all is said and done what is 
- the command devs would use to do all the auto-formatting
- the command devs/ci would use to check all the formatting is correct

**Comment by hamidzr** [comment]
nit: can we set it so that it actually tells us what it is complaining about? 

DeepinScreenshot_select-area_20220915153352

it's good that the format can solve most (all?) of these issues but if this doesn't show what's wrong this could be confusing once the ci fails and reports with "i don't like the formatting" but doesn't show what's wrong

**Review by hamidzr — APPROVED** [review]
LGTM. 
I added a minor comment about prettier error reporting
It might help to check in with the team about the potential merge conflicts before landing this

**Comment by keita-determined** [comment]
[test-intg-downstream](https://circleci.com/gh/determined-ai/determined/904396?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link) fails because SaaS needs change too, but we will update SaaS soon, and wouldn't be a problem

**Comment by keita-determined** [comment]
> nit: can we set it so that it actually tells us what it is complaining about?
> 
> DeepinScreenshot_select-area_20220915153352
> 
> it's good that the format can solve most (all?) of these issues but if this doesn't show what's wrong this could be confusing once the ci fails and reports with "i don't like the formatting" but doesn't show what's wrong

I checked https://prettier.io/docs/en/cli.html if Prettier CLI has `verbose` option, but they don't have it unfortunately. Usually, running `npm run format` will solve `npm run format-check` issue

**Pull request merged**
